src/Sulu/Component/HttpCache/EventSubscriber/InvalidationSubscriber.php

```diff
@@ -217,6 +231,33 @@ private function invalidateDocumentUrls($document, $locale)
         }
     }
 
+    /**
+     * Invalidates all tags and categories from excerpt extension.
+     *
+     * @param ExtensionBehavior $document
+     */
+    private function invalidateDocumentExcerpt(ExtensionBehavior $document)
+    {
+        $extensionData = $document->getExtensionsData();
+        if (!isset($extensionData['excerpt'])) {
+            return;
+        }
+
+        $excerpt = $extensionData['excerpt'];
+
+        if (isset($excerpt['tags'])) {
+            foreach ($this->tagManager->resolveTagNames($excerpt['tags']) as $tag) {
+                $this->invalidationHandler->invalidateReference('tag', $tag);
```

Comment:
Why do we need to invalidate the entire tag? Wouldn't it be enough to only invalidate pages which have the resulting pages/snippets/contacts whatever?

Reply:
we want to invalidate pages which uses the tag/category (for example as a filter for queries - smart-content in this case).